test/templates/reproducible/.jhipster/Foo.json

```diff
@@ -0,0 +1,25 @@
+{
+    "name": "Foo",
+    "fields": [
+        {
+            "fieldName": "name",
+            "fieldType": "String"
+        },
+        {
+            "fieldName": "numberPatternRequiredTom",
```

Comment:
Maybe use just simply `numberPatternRequired` for better code readability? In #10925 I used suffix Tom because all fields are suffixed with Tom there.